src/encoded/static/components/navigation/components/SearchBar.js

```diff
@@ -175,16 +177,32 @@ export class SearchBar extends React.PureComponent {
         const { href, currentAction } = this.props;
         const lastQuery = searchFilters.searchQueryStringFromHref(href);
 
-        this.setState(function({ typedSearchQuery: currQueryStr }){
-            let typedSearchQuery = currQueryStr; // No change - default
-            if (SearchBar.hasInput(lastQuery) && !SearchBar.hasInput(currQueryStr)) {
-                // Replace new value entered with current search query in URL if new value is empty string
-                typedSearchQuery = lastQuery;
+        setTimeout(function () {
+            this.setState(function ({ typedSearchQuery: currQueryStr, isSearchItemTypeDropDownOpen }) {
+                let typedSearchQuery = currQueryStr; // No change - default
+                if (SearchBar.hasInput(lastQuery) && !SearchBar.hasInput(currQueryStr)) {
+                    // Replace new value entered with current search query in URL if new value is empty string
+                    typedSearchQuery = lastQuery;
+                }
+                // Prevent closing onBlur if on selection view, or if have input.
+                const isVisible = isSelectAction(currentAction) || SearchBar.hasInput(typedSearchQuery) || isSearchItemTypeDropDownOpen || false;
+                return { typedSearchQuery, isVisible, isSearchItemTypeDropDownOpen: isVisible ? isSearchItemTypeDropDownOpen : false };
+            });
+        }.bind(this), 100);
```

Comment:
Using an arrow function `(params)=>{ .. }` is preferable to `function(params){ .. }.bind(this)`

Reply:
bind(this) is converted to arrow functions.